src/TreeSitter/Node.hs

```diff
@@ -20,70 +26,107 @@ data Node = Node
   , nodeNamedChildCount :: !Int32
   , nodeChildCount :: !Int32
   }
-  deriving (Show, Eq, Generic, CStorable)
+  deriving (Show, Eq, Generic)
 
 data TSPoint = TSPoint { pointRow :: !Int32, pointColumn :: !Int32 }
-  deriving (Show, Eq, Generic, CStorable)
-
+  deriving (Show, Eq, Generic)
 
 data TSNode = TSNode !(Ptr ()) !Int32 !Int32 !Int32
-  deriving (Show, Eq, Generic, CStorable)
+  deriving (Show, Eq, Generic)
+
+
+-- | 'Struct' is a strict 'Monad' with automatic alignment & advancing, & inferred type.
+newtype Struct a = Struct { runStruct :: forall b . Ptr b -> IO (a, Ptr a) }
```

Comment:
I got sick of managing pointers explicitly in the `Storable` instances, so I made a DSL to do it for me. Should not impact performance as the combinators evaluate strictly and are all inlined.